cmd/libsnap-confine-private/bpf-support.h

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (C) 2021 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef SNAP_CONFINE_BPF_SUPPORT_H
+#define SNAP_CONFINE_BPF_SUPPORT_H
+
+#include <linux/bpf.h>
+#include <stddef.h>
+
+/**
+ * bpf_pin_to_path pins an object referenced by fd to a path under a bpffs
+ * mount.
+ */
+int bpf_pin_to_path(int fd, const char *path);
+
+/**
+ * bpf_get_by_path obtains the file handle to the object referenced by a path
+ * under bpffs filesystem.
+ */
+int bpf_get_by_path(const char *path);
+
+/**
+ * bpf_load_prog loads a given BPF program.
+ *
+ * Passing non-NULL log buf, will populate the buffer with output from verifier
+ * if the program is found to be invalid.
+ */
+int bpf_load_prog(enum bpf_prog_type type, const struct bpf_insn *insns, size_t insns_cnt, char *log_buf,
+                  size_t log_buf_size);
+
+int bpf_prog_attach(enum bpf_attach_type type, int cgroup_fd, int prog_fd);
+
+/**
+ * bf_create_map creates a BPF and returns a file descriptor handle to it.
```

Comment:
nitpick, but this isn't quite specific enough though is it, AIUI BPF maps are something specific that is separate from the program in that maps are used to maintain state across multiple runs of a BPF and also are used to communicate state from userspace to the BPF function, so saying that it's just a BPF feels inadequate

Reply:
hmm the comment should state that a `BPF map` gets created, looks like I lost `map` somewhere along the way